src/main/kotlin/org/misarch/catalog/graphql/model/ProductVariant.kt

```diff
@@ -0,0 +1,60 @@
+package org.misarch.catalog.graphql.model
+
+import com.expediagroup.graphql.generator.annotations.GraphQLDescription
+import com.expediagroup.graphql.generator.annotations.GraphQLIgnore
+import kotlinx.coroutines.reactor.awaitSingle
+import org.misarch.catalog.graphql.model.connection.ProductVariantVersionConnection
+import org.misarch.catalog.graphql.model.connection.ProductVariantVersionOrder
+import org.misarch.catalog.persistance.model.ProductVariantVersionEntity
+import org.misarch.catalog.persistance.repository.ProductRepository
+import org.misarch.catalog.persistance.repository.ProductVariantVersionRepository
+import org.springframework.beans.factory.annotation.Autowired
+import java.util.*
+
+@GraphQLDescription("A variant of a Product.")
+class ProductVariant(
+    id: UUID,
+    @GraphQLDescription("If true, the ProductVariant is visible to customers.")
+    val isPubliclyVisible: Boolean, private val productId: UUID, private val currentVersion: UUID
```

Comment:
By the way, I just noticed that we also need an optional `imageURL` here, as you should have an intuitive guess how it might look like
(Compare with amazon: Small 48×48 icon (or something like that) showcasing how it looks like, or simply a color for example),
and an optional price as they can differ across variants.

Reply:
not sure yet if this is the job of this service, as we will have a separate media service
I think we need to discuss where we have which information regarding media

Reply:
I didn't say anything about storing the image itself, simply an image link that the frontend can use to display an image of the category.

Reply:
Still I would omit this for this PR, as it is currently unclear how we want to handle images exactly.
For example, will this return an URL, or are we just providing an id, ...